src/hocon_schema.erl

```diff
@@ -327,7 +327,7 @@ find_structs(Schema) ->
     RootNs = hocon_schema:namespace(Schema),
     {RootNs, RootFields, [
         {Ns, Name, Fields}
-     || {{Ns, Name}, Fields} <- lists:keysort(1, maps:to_list(All))
+     || {{_Schema, Ns, Name}, Fields} <- lists:keysort(1, maps:to_list(All))
```

Comment:
Fix the problem with a customized sort function instead.

Reply:
Or: define namespace for all!

Reply:
I think defining a namespace for every schema module is too strict.
we should allow an undefined namespace.

Reply:
I already defined the namespace for all here https://github.com/emqx/emqx/pull/7742
But we should also allow if the namespace is the same, the sort by schema.